api/envoy/extensions/filters/network/meta_protocol_proxy/v3/meta_protocol_proxy.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.meta_protocol_proxy.v3;
+
+import "envoy/config/core/v3/config_source.proto";
+import "envoy/config/core/v3/extension.proto";
+import "envoy/extensions/filters/network/meta_protocol_proxy/v3/route.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.meta_protocol_proxy.v3";
+option java_outer_classname = "MetaProtocolProxyProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Meta Protocol proxy]
+// Meta Protocol proxy.
+
+// [#not-implemented-hide:]
+// [#next-free-field: 7]
+message MetaProtocolProxy {
+  // The human readable prefix to use when emitting statistics.
+  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
+
+  // The name of the application protocol built on top of meta protocol.
+  string application_protocol = 2 [(validate.rules).string = {min_len: 1}];
+
+  oneof route_specifier {
+    option (validate.required) = true;
+
+    // The meta protocol proxies route table will be dynamically loaded via the Meta RDS API.
+    MetaRds rds = 3;
+
+    // The route table for the meta protocol proxy is static and is specified in this property.
```

Comment:
You could potentially put examples here, or in the proto docs for RouteConfiguration.

Reply:
An example of the meta protocol route configuration with matcher API has been added to the route.proto.